src/main/java/com/example/backend/scheme/service/SchemeService.java

```diff
@@ -0,0 +1,120 @@
+package com.example.backend.scheme.service;
+
+import com.example.backend.common.ErrorCode;
+import com.example.backend.exception.exceptions.EntityNotFoundException;
+import com.example.backend.exception.exceptions.IllegalInputException;
+import com.example.backend.scheme.dto.SchemeDto;
+import com.example.backend.scheme.model.Scheme;
+import com.example.backend.scheme.model.SchemeToCar;
+import com.example.backend.scheme.repository.SchemeRepository;
+import com.example.backend.scheme.repository.SchemeToCarRepository;
+import com.example.backend.user.model.User;
+import com.example.backend.user.service.UserService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.persistence.EntityManager;
+import jakarta.transaction.Transactional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.example.backend.common.Strings.isNullOrEmpty;
+
+@Service
+public class SchemeService {
+
+    private final EntityManager em;
+    private final SchemeRepository schemeRepository;
+    private final SchemeToCarRepository schemeToCarRepository;
+    private final UserService userService;
+    private static final String EMPTY_JSON_OBJECT = "{}";
+
+    public SchemeService(EntityManager em, SchemeRepository schemeRepository, SchemeToCarRepository schemeToCarRepository, UserService userService) {
+        this.em = em;
+        this.schemeRepository = schemeRepository;
+        this.schemeToCarRepository = schemeToCarRepository;
+        this.userService = userService;
+    }
+
+    public List<Scheme> list(String email, Pageable pageable) {
+        return schemeRepository.findAllByUserEmail(email, pageable);
+    }
+
+    public Scheme getById(Long id, String email) throws EntityNotFoundException {
```

Comment:
Czy my będziemy w końcu pozbywać się emaila? Bo obecnie po nazwie metody obstawiałbym, że powinienem jako argument podać tylko id.

Reply:
Musisz podać obiektu, na którym chcesz operować. Email jest wstrzykiwany automatycznie po tokenie JWT. Służy to temu, żeby nie dało się wyciągnąć danych innego użytkownika